lib/Data/ContentTypeData.php

```diff
@@ -44,4 +44,25 @@ public function addFieldDefinitionData(FieldDefinitionData $fieldDefinitionData)
     {
         $this->fieldDefinitionsData[] = $fieldDefinitionData;
     }
+
+    /**
+     * Sort $this->fieldDefinitionsData first by position, then by identifier.
+     */
+    public function sortFieldDefinitions()
+    {
+        usort(
+            $this->fieldDefinitionsData,
+            function ($a, $b) {
+                if ($a->fieldDefinition->position === $b->fieldDefinition->position) {
+                    if ($a->fieldDefinition->identifier === $b->fieldDefinition->identifier) {
```

Comment:
This cannot happen as we can't have the same field definition identifier within the same ContentType